src/Fhir.Anonymizer.Tool/FhirResourceDataProcessor.cs

```diff
@@ -74,33 +77,32 @@ public void AnonymizeBulkDataFolder(string inputFolder, string outputFolder, boo
                     Directory.CreateDirectory(resourceOutputFolder);
                 }
 
-                var processedCount = 0;
-                var processedErrorCount = 0;
+                int completedCount = 0;
+                int failedCount = 0;
                 using (FileStream inputStream = new FileStream(bulkResourceFileName, FileMode.Open))
                 using (FileStream outputStream = new FileStream(bulkResourceOutputFileName, FileMode.Create))
                 {
-                    using StreamReader reader = new StreamReader(inputStream);
-                    using StreamWriter writer = new StreamWriter(outputStream);
+                    using FhirStreamReader reader = new FhirStreamReader(inputStream);
+                    using FhirStreamConsumer consumer = new FhirStreamConsumer(outputStream);
+                    Func<string, string> anonymizeFunction = (content) => _engine.AnonymizeJson(content);
+
+                    Stopwatch stopWatch = new Stopwatch();
+                    stopWatch.Start();
 
-                    string resourceLine;
-                    string resultLine;
-                    while ((resourceLine = reader.ReadLine()) != null)
+                    FhirPartitionedExecutor executor = new FhirPartitionedExecutor(reader, consumer, anonymizeFunction);
```

Comment:
Will multiple threads (consumers) be writing to same file? If so, how do we ensure the correct order of the output?

Reply:
please ignore, i got the ordering logic :)